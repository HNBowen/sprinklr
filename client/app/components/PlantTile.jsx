import React from 'react'
import PropTypes from 'prop-types'

const PlantTile = (props) => {
  return (
      <div>
        <button onClick={function() { props.handleDelete(props.plant.id) }}>x</button>
        <img src={props.plant.image} onClick={function() { props.handleClick(props.plant.id) }}/>
        <span>{props.plant.name}</span>
        <span>Last watered: {props.plant.lastWatered}</span>
      </div>
    )
}

PlantTile.propTypes = {
  plant: PropTypes.object.isRequired,
  handleClick: PropTypes.func.isRequired
}

export default PlantTile