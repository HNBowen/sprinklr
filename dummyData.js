module.exports.existingPlants = [
  {
    name: "Peace Lily",
    image: "https://cdn.nurserylive.com/images/stories/virtuemart/product/resized/nurserylive-Peace-Lily,-Spathiphyllum-Plant_128x128.jpg",
    lastWatered: new Date(2017, 11, 20).toDateString(),
    dateAdded: new Date(2017, 8, 31).toDateString(),
    id: 1,
    user: "test_user_1"
  },
  {
    name: "Monstera",
    image: "https://static.wixstatic.com/media/e7be30_b5577954ef5347409f7c4ebaeadce8ce~mv2_d_4288_2848_s_4_2.jpg/v1/fill/w_256,h_256,q_85,usm_0.66_1.00_0.01/e7be30_b5577954ef5347409f7c4ebaeadce8ce~mv2_d_4288_2848_s_4_2.jpg",
    lastWatered: new Date(2017, 11, 21).toDateString(),
    dateAdded: new Date(2017, 7, 30).toDateString(),
    id: 2,
    user: "test_user_1"
  },
  {
    name: "Elephant Ear",
    image: "https://d13z1xw8270sfc.cloudfront.net/resize/68491/colocasia_esculenta_2.jpg/128/128/0/",
    lastWatered: new Date(2017, 10, 30).toDateString(),
    dateAdded: new Date(2017, 6, 30).toDateString(),
    user: "test_user_2"
  },
  {
    name: "Rabbit's Tail Fern",
    image: "https://cdn.nurserylive.com/images/stories/virtuemart/product/resized/nurserylive-small-rabbit-foot-fern-1_128x128.png",
    lastWatered: new Date(2017, 11, 15).toDateString(),
    dateAdded: new Date(2017, 6, 30).toDateString(),
    user: "test_user_2"
  }
];

module.exports.plantsToAdd = [
  {
    name: "Pothos",
    image: "https://images-na.ssl-images-amazon.com/images/I/71C+qx6FiUL._CR412,0,1060,1060_UX128.jpg",
    lastWatered: new Date().toDateString(),
    id: 3
  },
  {
    name: "Fiddle Leaf",
    image: "https://i.pinimg.com/236x/6f/d3/2e/6fd32e64735e460852ec3c507df10354.jpg",
    lastWatered: new Date().toDateString(),
    id: 4
  }
];

module.exports.users = [
  {
    name: "test_user_1",
    password: "test_user_1_password"
  },
  {
    name: "test_user_2",
    password: "test_user_2_password"
  }
]

