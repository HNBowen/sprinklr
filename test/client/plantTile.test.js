import React from 'react';
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import PlantTile from '../../client/app/components/PlantTile.jsx'

import {existingPlants} from '../../dummyData.js'

describe('PlantTile', function() {
    
  const plant = existingPlants[0]
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(<PlantTile plant={plant} handleClick={jest.fn()} handleDelete={jest.fn()} />)
  })

  it('should render correctly', function() {
    jestExpect(shallowToJson(wrapper)).toMatchSnapshot();
  })

  it('should render an image', function() {
    expect(wrapper.find('img')).to.have.length(1);
  })

  it('should render a span for title and last watered', function() {
    expect(wrapper.find('span')).to.have.length(2);
  })

  it('should render image source from props', function() {
    expect(wrapper.find('img').prop('src')).to.equal(plant.image);
  })

  it('should render name from props', function() {
    expect(wrapper.find('span').at(0).text()).to.equal(plant.name)
  })

  it('should render lastWatered from props', function() {
    expect(wrapper.find('span').at(1).text()).to.equal("Last watered: " + plant.lastWatered)
  })

  //it should call waterPlant function on click
  it('should call the waterPlant function when the image clicked', function() {
    const waterSpy = jest.fn();
    wrapper = shallow(<PlantTile plant={plant} handleClick={waterSpy} handleDelete={jest.fn()}/>)
    wrapper.find("img").simulate("click");

    expect(waterSpy.mock.calls.length).to.equal(1);
  })

  it('should call the waterPlant function with the plantId as the only argument', function(){
    const waterSpy = jest.fn();
    wrapper = shallow(<PlantTile plant={plant} handleClick={waterSpy} handleDelete={jest.fn()}/>);

    wrapper.find("img").simulate("click");

    expect(waterSpy.mock.calls[0][0]).to.equal(plant.id)
  })

  it('should call the handleDelete function when the x button is clicked', function() {
    const deleteSpy = jest.fn();
    wrapper = shallow(<PlantTile plant={plant} handleClick={jest.fn()} handleDelete={deleteSpy} />)

    wrapper.find("button").simulate("click")
    expect(deleteSpy.mock.calls.length).to.equal(1)
    expect(deleteSpy.mock.calls[0][0]).to.equal(plant.id)
  })

  //it should change class on click
  
})