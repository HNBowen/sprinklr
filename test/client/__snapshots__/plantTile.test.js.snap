// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantTile should render correctly 1`] = `
<div>
  <button>
    x
  </button>
  <img
    onClick={[Function]}
    src="https://cdn.nurserylive.com/images/stories/virtuemart/product/resized/nurserylive-Peace-Lily,-Spathiphyllum-Plant_128x128.jpg"
  />
  <span>
    Peace Lily
  </span>
  <span>
    Last watered: 
    Wed Dec 20 2017
  </span>
</div>
`;
